### DISCUSSION TITLE
Storing dynamic JObject in database

### DISCUSSION THREAD

**Discussion by LordofTurtles** [ravendb/ravendb#12920] [Q&A]
Hello

I have an application that does not have access to the domain models of my database, but does want to do small data changes on the stored data.
To do this, I am loading documents as dynamics. This works perfectly fine, and I get all the data correctly. When I edit these dynamic and save changes on the session, these are getting persisted.

However where I am getting stumped is when I'm trying to edit the object via a text JSON editor, and store the changed document, Raven will not store my changes.
My code is as follows:

            var form = JsonConvert.DeserializeObject<dynamic>(_formJson);      
            session.Store(form, (string)form?.Id);
            session.SaveChanges();

Inside the dynamic is a metadata property defining to which collection the document should belong. However no changes are being stored in the database.

Additionally I am trying to create a new object which is a documentof an existing document
Code:

           var deepCopyForm = JsonConvert.DeserializeObject<dynamic>(JsonConvert.SerializeObject(form));
           deepCopyForm.Id = null;
           deepCopyForm.@metadata.@id = null;
           session.Store(deepCopyForm);
           session.SaveChanges();

This is not storing a new object in my database, however it does create or increment a Hilo entry for JObjects strangely enough.

Is there a better/correct way to handle storing dynamic objects in your database?
Is the intended solution to ensure that you maintain the session entity tracking and not create a new object by deserializing the json representation of the document?

**Comment by haludi** [comment]
Hi,

Can you submit a failing test to demonstrate your issue?
https://ravendb.net/docs/article-page/4.2/csharp/start/test-driver

Best regards,
Igal

**Comment by ayende** [comment]
The underlying reason is here: `JsonConvert.DeserializeObject<dynamic>`

This is actually translated to:  `JsonConvert.DeserializeObject<object>`

But `object` has no properties, so you have an empty value there.

You can use `JObject`, instead, which will work.
